src/pages/profilePage/UserProfile.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from 'react';
+import useUserFormik from './utility/formikConfig';
+import photoUpload from '../images/photoUpload.svg';
+import ButtonUserPage from '../../components/button/UserPageButton';
+import { UserProfile as UserProfileType } from '../../types/components/componentType';
+import { handlePhotoChange } from './utility/photoUtils';
+import './userProfile.scss';
+
+const UserProfile: React.FC = () => {
+  const [userProfile, setUserProfile] = useState<UserProfileType | null>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [formError, setFormError] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Симулюємо отримання даних зі статичних об'єктів
+    const fetchUserProfile = async () => {
+      try {
+        setIsLoading(true);
+
+        // Затримка для симуляції часу відповіді від сервера
+        await new Promise((resolve) => setTimeout(resolve, 500));
+
+        // Тимчасові дані
+        const tempData: UserProfileType = {
+          profilePhoto: 'path/to/default/photo.jpg',
+          username: 'Liliia Virsta',
+          email: 'liliia@example.com',
+          dateOfBirth: '2000-01-01',
+          registrationDate: '2022-01-01',
+          lastLoginTimestamp: '2022-02-01',
+        };
+
+        setUserProfile(tempData);
+        setError(null);
+      } catch (err) {
+        setError('Error fetching user profile');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchUserProfile();
+  }, []);
+
+  const formik = useUserFormik({
+    userProfile,
+    formError,
+    setUserProfile,
+    setIsEditing,
+    setFormError,
+  });
+
+  const handleEditClick = () => {
+    setIsEditing(!isEditing);
+    formik.resetForm();
+  };
+
+  const handleLogout = () => {
+    // Виклик сервісу для виходу (з видаленням токенів)
+    localStorage.removeItem('access_token');
+    localStorage.removeItem('refresh_token');
+    setUserProfile(null);
+    setIsEditing(false);
+  };
+
+  const handleAddPhoto = () => {
+    const fileInput = document.createElement('input');
+    fileInput.type = 'file';
+    fileInput.accept = 'image/png, image/jpeg';
+    fileInput.addEventListener('change', () =>
+      handlePhotoChange(fileInput, formik.setFieldValue),
+    );
+    fileInput.click();
+  };
+
+  return (
+    <div className="container-profile">
+      <form className="form" onSubmit={formik.handleSubmit}>
+        {isLoading ? (
+          <p>Loading user profile...</p>
+        ) : userProfile ? (
+          <div className="profile-width">
+            <h1 className="heading--h1">Personal information</h1>
+            {isEditing ? (
```

Comment:
What does this logic mean? If you get successfully data in block **try**, the user will see text on page: 'Loading user profile'? And when we will have error in block **catch**, we will see the data?

Reply:
No, this is not accurate. Here's the breakdown:
While the data is being fetched (isLoading is true), it will display "Loading user profile..."
If the data fetching is successful (userProfile exists), it will display the user's profile information.
If there is an error during data fetching, it will display the error message.
If none of the above conditions are met, it will display "No user profile available."